C111 dependency injection

HadarRose opened the pull request:
- Added dependency injection
- changed error reporting to a more general report

Commit:
Added dependency injection

I added basic injection and everything seems to work but I can't help being terrified that somehow, somewhere, this code has been messed up beyond recognition.

Commit:
Changed error reporting to general errors

amanduggal commented on src/main/java/bootcamp/configuration/ConfigurationModule.java:
```diff
@@ -0,0 +1,44 @@
+                .setOAuthAccessToken(twitterKeys.getOauthaccesstoken())
+                .setOAuthAccessTokenSecret(twitterKeys.getOauthaccesstokensecret());
+        TwitterFactory tf = new TwitterFactory(cb.build());
+        twitter = tf.getInstance();
```
why make the twitter instance here instead of the provider method?

amanduggal commented on src/main/java/bootcamp/configuration/ConfigurationModule.java:
```diff
@@ -0,0 +1,44 @@
+@Module
+public class ConfigurationModule {
+    private static Logger logger = LoggerFactory.getLogger(ConfigurationModule.class);
+    private Twitter twitter;
```
you shouldn't need this. the provider method returns `Twitter`.

amanduggal commented on src/main/java/bootcamp/services/twitter4j/TwitterResourceService.java:
```diff
@@ -1,73 +1,35 @@
 import java.util.Optional;
 import java.util.stream.Collectors;
 
+@Module
```
not sure why you made this into a `module`. Being an injectable service should be sufficient here.

amanduggal commented on src/main/java/bootcamp/services/twitter4j/TwitterResourceService.java:
```diff
@@ -82,41 +44,45 @@ public Tweet statusToTweet(Status status) {
      * @param keyword word by which the timeline will be filtered
      * @return Optional<List < Tweet>> list of Tweet objects from the user's timeline, filtered by keyword
-     * @throws TwitterResourceException
+     * @throws TwitterResourceExceptionHandler
```
this isn't correct.

Commit:
Small style changes

-moved ConfigurationModule's functionality from constructor to provides method
- cleaned some comments

Commit:
Minor changes

removed module annotation where it was unneeded

Commit:
moved module to base directory
